### PR TITLE
Add AnimalTameEvent for handling animal taming

### PR DESCRIPTION
**Pull request by bs2609**
This PR adds a new `AnimalTameEvent`, allowing mods to handle animals being tamed.

There was some talk about possible use for something similar in #3046.
Here we just provide an event for mods to subscribe to, rather than adding new methods to `EntityTameable`.

Points to note:
*  The class used to represent the entity involved is `EntityAnimal`.
This is the class that is common to all vanilla animals that can be tamed.
The `EntityPlayer` performing the taming is also available through the event.
*  It supports horses and llamas as well, despite them not extending `EntityTameable`.
The behaviour is inherited from the AI handling of `AbstractHorse`.
*  An event means mods can prevent the taming of animals without interfering with other interactions.
Currently, you would have to anticipate the behaviour of `processInteract()` and then selectively abort the entire interaction.
*  The event is fired via patches to:
    *  `EntityAIRunAroundLikeCrazy.updateTask()`  (for vanilla `AbstractHorse` based entities)
    *  `EntityOcelot.processInteract()`
    *  `EntityWolf.processInteract()`
*  Mods will have to fire this event manually for other `EntityTameable` derived entities as well as other custom animals.

A quick test mod is included that makes wolves untameable.

### CODE REVIEW

File: src/test/java/net/minecraftforge/test/AnimalTameEventTest.java [hunk 1]
```diff
@@ -0,0 +1,25 @@
+package net.minecraftforge.test;
+
+import net.minecraft.entity.passive.EntityWolf;
+import net.minecraftforge.event.entity.living.AnimalTameEvent;
+import net.minecraftforge.fml.common.Mod;
+import net.minecraftforge.fml.common.eventhandler.SubscribeEvent;
+
+@Mod(modid = AnimalTameEventTest.MOD_ID, name = "AnimalTameEvent test mod", version = "1.0")
+@Mod.EventBusSubscriber
```
Comment (liach) [inline src/test/java/net/minecraftforge/test/AnimalTameEventTest.java:9]: This does not work in test mods according to @mezz. Register manually.

**Review by liach — CHANGES_REQUESTED** [review]

File: src/test/java/net/minecraftforge/test/AnimalTameEventTest.java (hunk 1 above)
Comment (Madsthunder) [reply src/test/java/net/minecraftforge/test/AnimalTameEventTest.java:9]: Yes, it does, you just have to ensure that `onAnimalTame` is a `static` method, which it is currently.

Comment (liach) [reply src/test/java/net/minecraftforge/test/AnimalTameEventTest.java:9]: Turned out @mezz was confused. :trollface:

**Review by liach — APPROVED** [review]
:trollface:

File: src/test/java/net/minecraftforge/test/AnimalTameEventTest.java (hunk 1 above)
Comment (mezz) [reply src/test/java/net/minecraftforge/test/AnimalTameEventTest.java:9]: Yeah it just needs to be `static`. Here is that comment: https://github.com/MinecraftForge/MinecraftForge/pull/3686#issuecomment-284681329

**Comment by mezz** [comment]
Tested and looks good.

**Pull request merged**